app/scripts/controllers/buildConfig.js

```diff
@@ -65,6 +66,40 @@ angular.module('openshiftConsole')
               });
             }
 
+            $scope.updatedBuildConfig = angular.copy($scope.buildConfig);
+            $scope.buildStrategy = $filter('buildStrategy')($scope.updatedBuildConfig);
+            $scope.strategyType = $scope.buildConfig.spec.strategy.type;
+            if(!$scope.buildStrategy.env) {
+              $scope.buildStrategy.env = [];
+            }
+            $scope.envVars = $scope.buildStrategy.env;
+
+            $scope.saveEnvVars = function() {
+              $scope.envVars = _.reject(  $scope.envVars, function(env) {
+                return env.name === "";
+              });
+              buildStrategy($scope.updatedBuildConfig).env = $scope.envVars;
+
+              DataService
+                .update("buildconfigs", $routeParams.buildconfig, angular.copy($scope.updatedBuildConfig), context)
+                .then(function success(){
+                  // TODO:  de-duplicate success and error messages.
+                  // as it stands, multiple messages appear based on how edit
+                  // is made.
+                  $scope.alerts['saveBCEnvVarsSuccess'] = {
+                    type: "success",
+                    // TODO:  improve success alert
+                    message: $scope.buildConfigName + " was updated."
+                  };
+                }, function error(){
+                  $scope.alerts['saveBCEnvVarsError'] = {
+                    type: "error",
+                    // TODO:  improve error alert
```

Comment:
we have a helper filter already for extracting failures from the response, go ahead and implement this TODO please